pkg/controllers/apply_controller.go

```diff
@@ -283,17 +294,17 @@ func mergeMapOverrideWithDst(src, dst map[string]string) map[string]string {
 	return r
 }
 
-// insertOwnerReference inserts a new owner
-func insertOwnerReference(owners []metav1.OwnerReference, newOwner metav1.OwnerReference) []metav1.OwnerReference {
-	found := false
+func findOwnerReference(owners []metav1.OwnerReference, target metav1.OwnerReference) bool {
```

Comment:
please add a comment for the function

Reply:
TODO: move this to a util dir or find an existing lib

Reply:
I have added a comment to this function.